packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -1616,16 +1593,16 @@ export class ContainerRuntime extends TypedEventEmitter<IContainerRuntimeEvents>
     private processFluidDataStoreOp(message: ISequencedDocumentMessage, local: boolean, localMessageMetadata: unknown) {
         const envelope = message.contents as IEnvelope;
         const transformed = { ...message, contents: envelope.contents };
-        const context = this.getContext(envelope.address);
+        const context = this.datastoreContexts.get(envelope.address);
         context.process(transformed, local, localMessageMetadata);
     }
 
     private bindFluidDataStore(fluidDataStoreRuntime: IFluidDataStoreChannel): void {
         this.verifyNotClosed();
-        assert(this.notBoundContexts.has(fluidDataStoreRuntime.id),
+        assert(this.datastoreContexts.notBoundContexts.has(fluidDataStoreRuntime.id),
             "Store to be bound should be in not bounded set");
-        this.notBoundContexts.delete(fluidDataStoreRuntime.id);
-        const context = this.getContext(fluidDataStoreRuntime.id) as LocalFluidDataStoreContext;
+        this.datastoreContexts.notBoundContexts.delete(fluidDataStoreRuntime.id);
+        const context = this.datastoreContexts.get(fluidDataStoreRuntime.id) as LocalFluidDataStoreContext;
```

Comment:
This asserts if it's not there... How are we so confident it's there already? And why is this the moment when we call `deferred.resolve(context)` below, as opposed to whenever the context itself was first added to the map?

Reply:
Starting to figure it out.  See my comment below in `DataStoreContexts.setupNew`.

We might still want a check up top to bail if the context isn't there at all yet, I've seen this assert get hit.